unstructured/partition/pptx.py

```diff
@@ -58,97 +67,235 @@ def partition_pptx(
-
-        for shape in _order_shapes(slide.shapes):
-            if shape.has_table:
-                table: pptx.table.Table = shape.table
-                html_table = convert_ms_office_table_to_text(table, as_html=True)
-                text_table = convert_ms_office_table_to_text(table, as_html=False).strip()
-                if text_table:
-                    metadata = ElementMetadata(
-                        filename=metadata_filename or filename,
-                        text_as_html=html_table,
-                        page_number=metadata.page_number,
-                        last_modified=metadata_last_modified or last_modification_date,
-                    )
-                    elements.append(Table(text=text_table, metadata=metadata))
-                continue
-            if not shape.has_text_frame:
-                continue
-            # NOTE(robinson) - avoid processing shapes that are not on the actual slide
-            # NOTE - skip check if no top or left position (shape displayed top left)
-            if (shape.top and shape.left) and (shape.top < 0 or shape.left < 0):
+
+class _PptxPartitioner:  # pyright: ignore[reportUnusedClass]
+    """Provides `.partition()` for PowerPoint 2007+ (.pptx) files."""
+
+    def __init__(
+        self,
+        file: Union[str, IO[bytes]],
+        # -- having default values for these arguments is not necessary for production uses because
+        # -- this object is always created by the classmethod. However it simplifies constructing
+        # -- this object in tests and makes them less sensitive to signature changes.
+        include_page_breaks: bool = True,
+        include_slide_notes: bool = False,
+        metadata_filename: Optional[str] = None,
+        metadata_last_modified: Optional[str] = None,
+    ) -> None:
+        self._file = file
+        self._include_page_breaks = include_page_breaks
+        self._include_slide_notes = include_slide_notes
+        self._metadata_filename = metadata_filename
+        self._metadata_last_modified = metadata_last_modified
+        self._page_counter = 0
+
+    @classmethod
+    def iter_presentation_elements(
+        cls,
+        file: Union[str, IO[bytes]],
+        include_page_breaks: bool,
+        include_slide_notes: bool,
+        metadata_filename: Optional[str],
+        metadata_last_modified: Optional[str],
+    ) -> Iterator[Element]:
+        """Partition MS Word documents (.docx format) into its document elements."""
+        return cls(
+            file,
+            include_page_breaks,
+            include_slide_notes,
+            metadata_filename,
+            metadata_last_modified,
+        )._iter_presentation_elements()
+
+    def _iter_presentation_elements(self) -> Iterator[Element]:
+        """Generate each document-element in presentation in document order."""
+
+        for slide in self._presentation.slides:
+            yield from self._increment_page_number()
```

Comment:
The yield pattern works well for creating specific element types and folding them into a generator. This may be less obvious to people new to the library or with less python experience. Could you add a bit more in-line documentation for how this pattern works?

Reply:
Sure, good idea :)

Reply:
Looks fantastic. Thanks